doc/tutorials/debug.rst

```diff
@@ -0,0 +1,371 @@
+or paused).
+
+.. figure:: images/debug_image7.png
+   :align: center
+
+   vcpu_list information
+
+vcpu_dumpreg
+============
+
+"vcpu_dumpreg vmid cpuid" provides vCPU related information such as
+registers values, etc.
+
+In the following example, we dump vCPU0 RIP register value and get into
+the SOS to search for the currently running function, using these
+commands::
+
+   cat /proc/kallsyms | grep RIP_value
+
+As you can see, vCPU0 is running in
+function ``acpi_idle_do_entry``.
+
+.. figure:: images/debug_image10.png
+   :align: center
+
+   vcpu_dumpreg information
+
+.. figure:: images/debug_image9.png
+   :align: center
+
+   system map information
+
+dumpmem
+=======
+
+"dumpmem mem_address" provides the specified memory target data such as
+the physical CPU (pCPU) number, etc.
+
+In this example, we know the pCPU active bitmap and physical CPU number
+physical memory address through
+``out/hv/apl-mrb-sbl/hypervisor/acrn.map``.
+Then we can dump the memory address of the pCPU active bitmap and CPU
+number, we will know that pCPU active bitmap is 0x000000000000000f and
+pCPU number is 0x0000000000000004.
+
+.. figure:: images/debug_image12.png
+   :align: center
+
+   dumpmem information
+
+.. figure:: images/debug_image11.png
+   :align: center
+
+   acrn map information
+
+dump_ioapic
+===========
+
+"dump_ioapic" provides IOAPIC information and we can get IRQ number,
+IRQ vector number, etc.
+
+.. figure:: images/debug_image14.png
+   :align: center
+
+   dump_ioapic information
+
+pt
+==
+
+"pt" provides pass-through detailed information, such as the virtual
+machine number, interrupt type, interrupt request, interrupt vector,
+trigger mode, etc.
+
+.. figure:: images/debug_image13.png
+   :align: center
+
+   pt information
+
+int
+===
+
+"int" provides interrupt information on all CPUs and their corresponding
+interrupt vector.
+
+.. figure:: images/debug_image17.png
+   :align: center
+
+   int information
+
+cpuid
+=====
+
+"cupid cpu_num" provides the highest EAX call parameter (leaf)
+supported by the CPU in the EAX register and the values of EBX, ECX, and EDX
+registers.
+
+.. figure:: images/debug_image15.png
+   :align: center
+
+   cpuid information
+
+rdmsr/wrmsr
+===========
+
+We can read and write model specific register (MSR) to get and set register
+values through ``rdmsr/wrmsr register_address``.
+
+In the following example, we can get IA32_APIC_BASE value through
+the command::
+
+   rdmsr 1b
+
+and see that 1B (Hexadecimal) is the IA32_APIC_BASE MSR address.
+
+.. figure:: images/debug_image16.png
+   :align: center
+
+   IA32_APIC_BASE register information
+
+.. figure:: images/debug_image18.png
+   :align: center
+
+   rdmsr information
+
+An example
+==========
+
+As an example, we'll show how to obtain the interrupts of a pass-through USB device.
+
+First, we can get the USB controller BDF number (0:15.0) through the
+following command::
+
+   lspci | grep "USB controller"
```

Comment:
I'm assuming this needs to be run in the SOS console, correct?

We need to clarify where this needs to be run from.